### ISSUE TITLE
Friendly request regarding enhancement of package manager in emacs

### ISSUE THREAD

**Issue by kirk86** [melpa/melpa#3649]
Dear fellow compatriots,

I am total newb when it comes to emacs and its crazy configuration. I would really like to delve more into it and use as my daily drive for my work but its crazy configuration is what keeps us "people like me newbs" away from it.

You see I really love the package manager and I think that is in the right direction into bringing emacs into the 21st century.

But the whole purpose of the package manger is being overshadowed by the fact that for every single package you install even if you just want to keep the simple default configurations, one way or another you would have to add some sort of configuration in `init.el` or `whatever the equivalent configuration file is`.

For instance if I install `evil` from melpa then in order to use it I would definitely have to add the following lines
`(require 'evil)`
`(evil-mode 1)`
into my configuration file. It would be nice if that could have been taken care of by the package manager and I am not talking about anything tailor made I'm just referring to the simple case of the default configuration to be added automatically when the package is installed via the package manager.

Do you guys have any idea how that can be accomplished?

Thank you in advance and please forgive my ingnorance.

**Comment by purcell** [comment]
In this specific case you can probably use the `M-x customize` interface to enable `evil-mode`, then save your customisations: this would normally work without you needing to put anything in your startup files.

When you install Emacs there's immediately a huge standard library of bundled packages available, and most of them provide functionality you will not want to be enabled automatically. Having a package available is therefore not the same as wanting it to be enabled, and externally-installable packages follow the same philosophy. Some amount of user configuration is therefore inevitable, either via `customize` or startup files.

I'll close this issue, since it isn't really a MELPA issue, and we don't maintain the package management part of Emacs, but feel free to ask further questions in comments if you have any.

**Comment by kirk86** [comment]
Thank you Purcell for your response,
may I ask just one clarification? Since I am new to this maybe I've misunderstood your point.
From what I understand `emacs` gives you the possibility to install any package from the available repositories and the default setting is that is not enabled? Correct?

Which is a bit different from the default behaviour someone would find in text editors like `sublime`, were when you install a package via the package manager and the default setting is for the package to be enabled after the installation.

But that's ok I guess.

For instance, let's take a concrete example, say that I install a package `M-x package-install RET`
`elpy RET`
and then I do `M-x customize` and  I enable `elpy`
does that mean that the configuration for `elpy` will have been added to my `init.el` for `elpy` to operate normally the next time I open and close emacs.

In other words if I look at `init.el` will I see something similar to
`(package-initialize)`
`(elpy-enable)`
if I do `M-x customize` and `enable elpy` through that emacs buffer interface?

Thank you.

**Comment by purcell** [comment]
> Thank you Purcell for your response,
> may I ask just one clarification? Since I am new to this maybe I've misunderstood your point.
> From what I understand emacs gives you the possibility to install any package from the available repositories and the default setting is that is not enabled? Correct?
> 

Yep.

> Which is a bit different from the default behaviour someone would find in text editors like sublime, were when you install a package via the package manager then the default setting is for the package to be enabled after the installation.
> 

Yes, it's different.


> For instance, let's take a concrete example, say that I install a package M-x package-install RET
> elpy RET
> and then I do M-x customize and I enable elpy
> does that mean that the configuration for elpy will have been added to my init.el for elpy to operate normally the next time I open and close emacs.

Not necessarily. If the package supports being enabled via the customise interface, then some settings will get saved to your custom file (e.g. `~/.emacs.d/custom.el`), allowing the settings to be restored at restart time. If the package does not support that (and it looks like `elpy` might be in that category) then you will _have_ to add code to your startup file. You should embrace this, though: you'll need to start doing it sooner or later, so why not dive in? ;-)
